### ISSUE TITLE
Ability to post files from a local directory instead of URL

### ISSUE THREAD

**Issue by brbester** [CiscoDevNet/ciscosparkapi#9]
more a feature request...

Here is how I had to implement in my legacy library using toolbelt... (https://pypi.python.org/pypi/requests-toolbelt)

```python
from requests_toolbelt.multipart.encoder import MultipartEncoder

def post_localfile(at, roomId, filename, text='', toPersonId='', toPersonEmail=''):
    openfile = open(filename, 'rb')
    filename = ntpath.basename(filename)
    payload = {'roomId': roomId, 'files': (filename, openfile, 'image/jpg')}
    if text:
        payload['text'] = text
    if toPersonId:
        payload['toPersonId'] = toPersonId
    if toPersonEmail:
        payload['toPersonEmail'] = toPersonEmail
    m = MultipartEncoder(fields=payload)
    headers = {'Authorization': _fix_at(at), 'Content-Type': m.content_type}
    resp = requests.request("POST",url=_url('/messages'), data=m, headers=headers)
    file_dict = json.loads(resp.text)
    file_dict['statuscode'] = str(resp.status_code)
return file_dict
```

**Comment by cmlccie** [comment]
Thanks for the example code!  This should be easy enough to add.  I'll add this to the backlog and see how best to add the functionality to the package.

-Thank you!

**Comment by cmlccie** [comment]
Brad,

Thanks again sharing your code on how you implemented this functionality.  I had to modify it slightly due to some funkiness that I saw with the 'content-type' associated with the file when doing the multi-part encoding.  _For example:_  When uploading a text file, with the ```'content-type'``` set to ```'image/jpg'```, the text file wouldn't display in the Spark client properly.

I did a little  searching and found the Python [mimetypes](https://docs.python.org/2/library/mimetypes.html) library module that has a ```guess_type()``` function that will attempt to guess the mime-type based on the file's extension.  It's probably not perfect, but it seems to do a good job, and I have set a default of ```'text/plain'``` if the ```guess_type()``` returns None.

This functionality is implemented in v0.3, which has been posted to PyPI.  Check it out and please let me know if you run into any issues.  See the [release notes](https://github.com/CiscoDevNet/ciscosparkapi/releases/tag/v0.3), for example code on uploading local files.
